From: Ari Chivukula <arichiv@chromium.org>
Date: Thu, 11 Aug 2022 00:39:04 +0000
Subject: Stop cross-origin cache hits

Currently, if a.com is loaded and has a favicon at a.com/icon.png and
then b.com is loaded and has the exact same favicon, the cache entry is
shared which permits b.com to notice that a.com was visited. The end
goal of this task is to prevent cross-origin cache leaks.

This CL integrates the new variant of GetFaviconIDForFaviconURL into
UpdateFaviconMappingsAndFetch so that we filter the results per-pageurl
based on the origin. This should fully resolve the task, although the db
itself isn't fully partitioned.

This CL is part of a series:
(1) Cache browser test
(2) Add new cache check function
(3) Stop cross-origin cache hits

Bug: 1300214
Change-Id: I9bf04982abea136a00e2b5252726a1cdef8e8550
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3822882
Reviewed-by: Scott Violet <sky@chromium.org>
Auto-Submit: Ari Chivukula <arichiv@chromium.org>
Commit-Queue: Ari Chivukula <arichiv@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1033798}

License: GPL-3.0-only - https://spdx.org/licenses/GPL-3.0-only.html
---
 components/favicon/core/favicon_backend.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/components/favicon/core/favicon_backend.cc b/components/favicon/core/favicon_backend.cc
--- a/components/favicon/core/favicon_backend.cc
+++ b/components/favicon/core/favicon_backend.cc
@@ -808,4 +808,5 @@ bool FaviconBackend::ClearAllExcept(const std::vector<GURL>& kept_page_urls) {
   db_->BeginTransaction();
   return true;
 }
+// DROP THIS PATCH
 }  // namespace favicon
--
2.25.1
